src/RNTuple/Writing/TFileWriter.jl

```diff
@@ -0,0 +1,565 @@
+using StaticArrays
+using UnROOT
+using UnROOT: RNTupleFrame, ClusterSummary, PageDescription
+using XXHashNative: xxh3_64
+using Accessors
+using Tables: istable, columntable, schema
+
+function color_diff(ary1, ary2)
+    if length(ary1) != length(ary2)
+        printstyled("!!! Length mismatch !!!: length(ary1)=$(length(ary1)), length(ary2)=$(length(ary2))\n", color=:red)
+    end
+    print("[")
+    x = 0
+    for (i,j) in zip(ary1, ary2)
+        if x % 8 == 0
+            print("    ")
+        end
+        if x % 16 == 0
+            println()
+            x = 0
+        end
+        if i != j
+            printstyled("$(repr(i))/$(repr(j)), ", color=:red)
+        else
+            printstyled("$(repr(i)), ", color=:green)
+        end
+        x += 1
+    end
+    println()
+    println("]")
+end
+
+function rnt_write(io::IO, x::AbstractString; legacy=false)
+    L = length(x)
+    if legacy
+        if L > typemax(UInt8)
+            error("String longer than 255 not implemented")
+        end
+        write(io, UInt8(L))
+        write(io, codeunits(x))
+    else
+        write(io, UInt32(L))
+        write(io, codeunits(x))
+    end
+end
+
+function rnt_write(io::IO, x::AbstractVector)
+    error("Ambiguous type: $(typeof(x)), use Write_RNTupleListFrame or similar wrapper")
+end
+
+function rnt_write(io::IO, x; legacy=false)
+    if legacy
+        write(io, bswap(x))
+    else
+        write(io, x)
+    end
+end
+
+function rnt_write(io::IO, x::AbstractVector{UInt8}; legacy=false)
+    if legacy
+        write(io, reverse(x))
+    else
+        write(io, x)
+    end
+end
+
+function test_io(obj, expected; kw...)
+    a = IOBuffer()
+    rnt_write(a, obj; kw...)
+    ours = take!(a)
+    if ours != expected
+        color_diff(ours, expected)
+    end
+end
+
+function rnt_write(io::IO, x::UnROOT.FileHeader32)
```

Comment:
Are these `legacy` flags something which should be propagated from the function call as keyword arguments?

Reply:
in this case we know `FileHeader32` will always be written out "legacy" (i.e. big endian) so we don't take keyword argument here.

Because it doesn't make sense to `rnt_write()` a `FileHeader32` with `legacy=false`.

Although maybe this should be called `bigendian=true|false`